tram_fsm/lib/tram_fsm.ex

```diff
@@ -34,23 +45,34 @@ defmodule TramFsm do
       iex> TramFsm.transition(pid, :move)
       {:ok, %{tram_state: :in_depot}}
   """
+  @spec start_link(tram_state() | none()) :: {:ok, pid()} | {:error, String.t()}
   def start_link(default \\ %{tram_state: :in_depot}) do
     GenServer.start_link(__MODULE__, default)
   end
 
   @impl true
+  @spec init(any()) :: {:ok, any()}
```

Comment:
`any()`?! 
```suggestion
  @spec init(map()) :: {:ok, map()}
```
Или даже более строго, `%{required(:tram_state) => atom()}`.

Reply:
поменял на `tram_state()`